from_cpython/Modules/socketmodule.c

```diff
@@ -3119,7 +3119,8 @@ sock_dealloc(PySocketSockObject *s)
 {
     if (s->sock_fd != -1)
         (void) SOCKETCLOSE(s->sock_fd);
-    Py_TYPE(s)->tp_free((PyObject *)s);
+    // Pyston change: the gc takes care of this
+    //Py_TYPE(s)->tp_free((PyObject *)s);
```

Comment:
yeah I think we should be able to deal with this.  the problem was that socket's had tp_free == PyObject_Del, which was failing for gc allocated (not untracked) allocations.  maybe we can just have it return without doing anything in the case of a gc-tracked allocation.